src/Aspire.Dashboard/Components/Layout/MainLayout.razor

```diff
@@ -9,7 +9,7 @@
     <FluentHeader>
         <div class="header-title">
             <FluentAnchor IconStart="@(new AspireIcons.Size32.Logo())" Appearance="Appearance.Stealth" Href="/" Class="logo">
-                @string.Format(Loc[nameof(Layout.MainLayoutDashboardName)], ResourceService.ApplicationName)
+                @DashboardClient.FormatApplicationName(Loc[nameof(Layout.MainLayoutDashboardName)], () => InvokeAsync(StateHasChanged))
```

Comment:
This could be moved to its own component to hide the work around refresh away. Also, a benefit of it being its own component is the StateHasChanged notification just updates that component rather than the entire page.

Reply:
This pattern is also used for `<PageTitle>` on almost every page, and I don't think that can be moved to a component. The refresh is only needed if the page loads before the client is connected, in which case more than just this anchor will be refreshed. I can't see a clean path forward here, but am open to suggestions.

Reply:
A page title is just a [component with a child render fragment](https://github.com/dotnet/aspnetcore/blob/d461781f80ef098b3d584d6397f3baa1d6b9f285/src/Components/Web/src/Head/PageTitle.cs#L18).

We can:

1. Create a simple component that wraps injecting dashboard client, calling FormatApplication name, and doing formatting.
2. Put the component inside page title, e.g. `<PageTitle><ApplicationName FormatString="Loc[nameof(Layout.MainLayoutDashboardName)]" /><PageTitle>`

Reply:
I've got this working locally, but will file a follow up PR so we can review the change independently.

Reply:
https://github.com/dotnet/aspire/pull/1555